packages/core/src/middleware/koa-check-admin.ts

```diff
@@ -0,0 +1,20 @@
+import { MiddlewareType } from 'koa';
+import { IRouterParamContext } from 'koa-router';
+
+import { hasAdminUsers } from '@/queries/user';
+
+export default function koaWelcomeProxy<
+  StateT,
+  ContextT extends IRouterParamContext,
+  ResponseBodyT
+>(): MiddlewareType<StateT, ContextT, ResponseBodyT> {
+  return async (ctx) => {
+    if (await hasAdminUsers()) {
```

Comment:
Hmmm, I'm just thinking, what if a user registered an admin account, and revoke the admin role? Then there would be a non-admin user in the DB, but `hasAdminUsers` will return false

Reply:
I think there should be a rule that one cannot change the role himself. Or even there should be a super admin (owner) of the tenant, whose role cannot be changed, or can only be passed to another user, like Slack workspace owner.

Reply:
Agree, what @demonzoo said can be a potential security issue.

Reply:
I might prefer to let the role control flow handle that.  Under current status, if all admin-account has been revoked, technically we should enable the register flow. 
 In another word, as you mentioned,  I think revoking all admin role action itself puts the whole system into an unstable status. Not checking `admin roles` and allowing register logic.

Reply:
Unless we put a more strict rule on this, only 0 users found, could one create a new admin account.

Reply:
I think this deserves a discussion. Mind creating a ticket and bringing it to the team meeting real quick?